BookStore.API/Controllers/BookController.cs

```diff
@@ -5,117 +5,69 @@
 namespace ApiBookStore.Controllers
 {
     [Route("api/[controller]")]
-    [ApiController]
-    public class BookController : ControllerBase
+    public class BookController : Controller
     {
+        private readonly BookService _bookService;
 
-        public BookController()
+        public BookController(BookService bookService)
         {
-
+            _bookService = bookService;
         }
 
-        // GET: api/Books
+        // GET: api/book
         [HttpGet]
-        public IEnumerable<Book> GetBooks()
+        public async Task<IActionResult> GetAll()
         {
-            return null;
+            var books = await _bookService.GetAll();
+            return Ok(books);
         }
 
-        // GET: api/Books/5
+        // GET: api/book/5
         [HttpGet("{id}")]
-        public async Task<IActionResult> GetBook([FromRoute] int id)
+        public async Task<IActionResult> GetById(string id)
         {
-            if (!ModelState.IsValid)
-            {
-                return BadRequest(ModelState);
-            }
-
-
-            if (id == null)
-            {
+            var book = await _bookService.GetById(id);
+            if (book == null)
                 return NotFound();
-            }
 
-            return Ok(null);
+            return Ok(book);
         }
 
-        // PUT: api/Books/5
-        [HttpPut("{id}")]
-        public async Task<IActionResult> PutBook([FromRoute] int id, [FromBody] Book book)
+        // POST: api/book
+        [HttpPost]
+        public async Task<IActionResult> Create([FromBody] Book book)
         {
-            if (!ModelState.IsValid)
-            {
-                return BadRequest(ModelState);
-            }
-
-            if (id != book.Id)
-            {
+            if (book == null)
                 return BadRequest();
-            }
 
-            //_context.Entry(book).State = Microsoft.EntityFrameworkCore.EntityState.Modified;
+            await _bookService.Create(book);
 
-            //try
-            //{
-            //    //await _context.SaveChangesAsync();
-            //}
-            //catch (DbUpdateConcurrencyException)
-            //{
-            //    if (!BookExists(id))
-            //    {
-            //        return NotFound();
-            //    }
-            //    else
-            //    {
-            //        throw;
-            //    }
-            //}
-
-            return NoContent();
+            return CreatedAtAction(nameof(GetById), new { id = book.Id }, book);
         }
 
-        // POST: api/Books
-        [HttpPost]
-        public async Task<IActionResult> PostBook([FromBody] Book book)
+        // PUT: api/book/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> Update(string id, [FromBody] Book book)
```

Comment:
You don't need to add id as an argument. Book has the id property that will be used for updating.